Feat: Repositioning by long press on robots

MLopezJ opened the pull request:
Addressing changes of this PR will allow admins to set new position and new rotation of robots on both desktop and mobile devices. 


---|Mobile|Desktop|
---|----|---
**Set New Rotation**|Press short time on robot|Click on robot
**Set New Position**|Press long time on robot|Double click on robot

MLopezJ committed:
feat: add double click event on Robot component

MLopezJ committed:
fix: set repositioning of robots with double click event

Previously, the repositioning of robots occurs with onPointerDown event. Now it is with onDoubleClick.

MLopezJ committed:
fix: set rotation of robot with gestures

Before this, as an admin you are just able to set the rotation using the mouse wheel, which represent an issue if the admin is using a mobile device.
Addressing those changes, the admin is going to be able to set the rotation similar than the user does on Game view.

MLopezJ committed:
refactor: upadate struct of 'selectedRobot' state

This way, that state will store not only the mac of the robot selected but the desired action as well.
Doing that, code is simplified removing 'defineAction' and 'resetAction' methods

MLopezJ committed:
fix: disable scrolling while rotating a robot

MLopezJ committed:
feat: allow re-positioning of robot by pressing for long time on it

Add a new behavior that allows the re-positioning of robots on the admin view by pressing on the robot for an amount of time bigger than 1.5 seconds. Below that time, the admin is going to be able to set the rotation of the robot by pressing on it, but abow that time the reposition of the robot is allowed. In this way, admin now is able to set new position and new rotation no matter the device they are using.

coderbyheart commented on src/app/pages/Admin.tsx:
```diff
@@ -107,8 +151,22 @@ export const Admin = () => {
-							onPointerDown={() => {
-								setSelectedRobot(mac)
+							onPointerDown={(args) => {
+								startTimer()
```
I would like to see a simpler API that hides the implementation detail:

```typescript
onPointerDown={(args) => {
  startLongPressDetection()
}}
onPointerUp={() => {
  if (endLongPressDetection()) {
    console.log(`Long pressed!`)
  } else {
    console.log(`Not long pressed!`)
  }
}}
```

MLopezJ replied:
I came with this implementation. 

``` JavaScript 

onPointerDown={(args) => {
    startLongPressDetection()
 }}

onPointerUp={() => {
   const isLongPressDetected = endLongPressDetection()
  
   isLongPressDetected
	  ? setSelectedRobot({ mac, action: 'reposition' })
	  : setSelectedRobot({ mac, action: 'rotation' })
}}

```

I think `isLongPressDetected` var add legibility to the code and for that I'm using it instead of using the method directly on the validation.

MLopezJ replied:
Other option could be to rename `endLongPressDetection` method to `isLongPressDetected` instead but maybe it will make the hook API a little bit ambiguous

coderbyheart replied:
Sounds good to me.

MLopezJ committed:
fix: simplify 'usePressTimer' API

Hide implementation details

MLopezJ committed:
refactor: rename 'usePressTimer' hook

from 'usePressTimer' to 'usePressDetection'

MLopezJ committed:
docs: explain admin actions

Add documentation to explain how to assign robots to teams, set new position and set new rotation

coderbyheart commented on src/hooks/usePressDetection.tsx:
```diff
@@ -0,0 +1,23 @@
+import { useRef } from 'react'
+
+export const usePressDetection = (): [() => void, () => boolean] => {
```
Needs test

MLopezJ replied:
Tests addressed. New provider created in order to make testing easier. `useDragGesture` hook was taken as example for that.

MLopezJ commented on README.md:
```diff
@@ -66,3 +66,30 @@ work correctly.
+from the teams options below the input field. After pick the robot team, the
+robot on the game map should turn the same color than the team.
+
+#### Positioning and rotation
```
Add documentation to inform about new features and changes done.

This can be improve adding images for example, but I think we can invest more time on this later and continue focusing on having a functional game by the moment .

MLopezJ committed:
fix: create new provider for 'pressDetection' hook

New provider was created with the intention of make easier the testing of the hook

MLopezJ committed:
test: create tests for 'usePressDetection' hook

coderbyheart committed:
docs: improve docs

coderbyheart committed:
fix: reduce time to detect long press
